DB/src/main/java/io/deephaven/db/tables/utils/csv/InferenceSpecs.java

```diff
@@ -0,0 +1,294 @@
+
+    /**
+     * The "standard" inference, does not parse floats or bytes.
+     *
+     * <p>
+     * Contains the following parsers:
+     *
+     * <ul>
+     * <li>{@link Parser#INSTANT}</li>
+     * <li>{@link Parser#SHORT}</li>
+     * <li>{@link Parser#INT}</li>
+     * <li>{@link Parser#LONG}</li>
+     * <li>{@link Parser#DOUBLE}</li>
+     * <li>{@link Parser#BOOL}</li>
+     * <li>{@link Parser#CHAR}</li>
+     * <li>{@link Parser#STRING}</li>
+     * <li>{@link Parser#BYTE}</li>
+     * <li>{@link Parser#FLOAT}</li>
+     * </ul>
+     *
+     * Uses the default {@link #onNullParser()}.
+     *
+     * <p>
+     * Note: the byte and float parsers are only relevant when the appropriate {@link #limitToType(Type)} is invoked.
+     *
+     * @return the standard inference
+     */
+    public static InferenceSpecs standard() {
+        return builder().addParsers(
+                Parser.INSTANT,
+                Parser.SHORT,
+                Parser.INT,
+                Parser.LONG,
+                Parser.DOUBLE,
+                Parser.BOOL,
+                Parser.CHAR,
+                Parser.STRING,
+                Parser.BYTE,
+                Parser.FLOAT)
+                .build();
+    }
+
+    /**
+     * The standard parsers with additional {@link java.time.Instant}-based parsing.
+     *
+     * <p>
+     * Contains the following parsers:
+     *
+     * <ul>
+     * <li>{@link Parser#INSTANT}</li>
+     * <li>{@link Parser#INSTANT_DB}</li>
+     * <li>{@link Parser#epochMilliAndMicroParsers21stCentury(Parser)}, with {@link Parser#LONG}</li>
+     * <li>{@link Parser#SHORT}</li>
+     * <li>{@link Parser#INT}</li>
+     * <li>{@link Parser#LONG}</li>
+     * <li>{@link Parser#DOUBLE}</li>
+     * <li>{@link Parser#BOOL}</li>
+     * <li>{@link Parser#CHAR}</li>
+     * <li>{@link Parser#STRING}</li>
+     * <li>{@link Parser#BYTE}</li>
+     * <li>{@link Parser#FLOAT}</li>
+     * </ul>
+     *
+     * Uses the default {@link #onNullParser()}.
+     *
+     * <p>
+     * Note: the byte and float parsers are only relevant when the appropriate {@link #limitToType(Type)} is invoked.
+     *
+     * @return the standard times inference
+     */
+    public static InferenceSpecs standardTimes() {
+        final List<Parser<Instant>> parsers = Parser.epochMilliAndMicroParsers21stCentury(Parser.LONG);
+        return builder().addParsers(
+                Parser.INSTANT,
+                Parser.INSTANT_DB,
+                parsers.get(0),
+                parsers.get(1),
+                Parser.SHORT,
+                Parser.INT,
+                Parser.LONG,
+                Parser.DOUBLE,
+                Parser.BOOL,
+                Parser.CHAR,
+                Parser.STRING,
+                Parser.BYTE,
+                Parser.FLOAT)
+                .build();
+    }
+
+    /**
+     * The parsers, in preference-based order.
+     *
+     * @return the parsers
+     */
+    public abstract List<Parser<?>> parsers();
```

Comment:
In some cases, we know that if one parser can parse a particular input, several other parsers also can (e.g. short implies int or long). We ought to try to be aware of this in order to short-circuit parsing in those cases.